source/falaise/snemo/datamodels/base_topology_pattern.h

```diff
@@ -0,0 +1,119 @@
+/// \file falaise/snemo/datamodels/base_topology_pattern.h
+/* Author(s) :    Steven Calvez <calvez@lal.in2p3.fr>
+ * Creation date: 2015-09-19
+ * Last modified: 2015-11-11
+ *
+ * Description: The base class of topology patterns
+ */
+
+#ifndef FALAISE_SNEMO_DATAMODEL_BASE_TOPOLOGY_PATTERN_H
+#define FALAISE_SNEMO_DATAMODEL_BASE_TOPOLOGY_PATTERN_H 1
+
+// Standard library:
+#include <map>
+#include <string>
+
+// Third party:
+// - Bayeux/datatools:
+#include <bayeux/datatools/i_serializable.h>
+#include <bayeux/datatools/i_tree_dump.h>
+
+// This project:
+#include <falaise/snemo/datamodels/base_topology_measurement.h>
+#include <falaise/snemo/datamodels/particle_track.h>
+
+namespace snemo {
+
+namespace datamodel {
+
+/// \brief The base class of reconstructed topology
+class base_topology_pattern : public datatools::i_serializable, public datatools::i_tree_dumpable {
```

Comment:
`base_topology` or `base_pattern` might be clearer names, having both seems redundant.

Reply:
I don't think so `base_topology_pattern` is the base class for `topology_2e_pattern`, `topology_1e_pattern`,...  It is just like the `base_trajectory_pattern` class